include/Reading.hpp

```diff
@@ -45,7 +45,7 @@ class ReadingIdentifier {
 	virtual ~ReadingIdentifier(){};
 
 	virtual size_t unparse(char *buffer, size_t n) = 0;
-	virtual bool operator==(ReadingIdentifier const &cmp) const;
+  bool operator==(ReadingIdentifier const &cmp) const;
```

Comment:
I don't understand the need for that. Can you paste the compiler error? I just tried with clang6 and gcc9 (don't have gcc8 installed)

Reply:
Even without compiler error, you probably do not want to shadow 
`Base::operator==(const Base&)` with `Derived::operator==(const Derived&)` I guess but intend to overload, right ? Actually I planned to submit a pull request, which adapts the design of this class, but I could not test because of the compilation error and the failing googletest built. So I created small requests for those first. Once these are fixed in master, I would suggest a change, which avoids the requirement of all derived classes to implement the comparison operator in the base class.